CRM/Core/BAO/CustomField.php

```diff
@@ -2112,7 +2112,7 @@ protected static function prepareCreate($params) {
         //don't insert only value separator as default value, CRM-4579
         $defaultValue = self::getOptionGroupDefault($params['option_group_id'], !empty($params['serialize']));
 
-        if (!CRM_Utils_System::isNull(explode(CRM_Core_DAO::VALUE_SEPARATOR, $defaultValue))) {
+        if (!CRM_Utils_System::isNull($defaultValue) && !CRM_Utils_System::isNull(explode(CRM_Core_DAO::VALUE_SEPARATOR, $defaultValue))) {
```

Comment:
Because CRM_Utils_System::isNull [is whack](https://github.com/civicrm/civicrm-core/blob/0b2e88334f8d27a737fa4061bd79926ab49a94db/tests/phpunit/CRM/Utils/SystemTest.php#L186), this may potentially evaluate differently with this change. I can't think of a working example at the moment but some variation on `$defaultValue = '' . CRM_Core_DAO::VALUE_SEPARATOR . 'null'` might expose it.

Is there a reason to add another instance of this function or can it just be `if ($defaultValue !== NULL && !CRM_Utils_System::isNull(explode(CRM_Core_DAO::VALUE_SEPARATOR, $defaultValue))) {`?

Reply:
I updated to that and that seems ok on php8.1